app/javascript/packs/search.js

```diff
@@ -1,24 +1,40 @@
 import List from 'list.js';
 
 $( function () {
+  let checkboxChangeFlag = false;
+  let dateChangeFlag = false;
+  let initialStartDate = parseInt($("#starting_year").val());
+  let initialEndDate = parseInt($("#ending_year").val());
+
   // Add .agency-separator to agency pipe separator
   $(".agency-names").html(function(_, html){
     return html.replace("|","<span class='agency-separator'>|</span>");
   });
 
   // Deselect all buttons
-  $(".clear-all").on('click', function(){
+  $(".clear-all").on('click', function(e){
+    e.preventDefault();
     const parentId = $(this).parent()[0].id; // "sections" or "agencies"
     // uncheck the checkboxes, fire the change event
-    $(`#${parentId} input:checkbox`).prop("checked", false).trigger("change")
+    let checked = $(`#${parentId} :checked`);
+    checked.each(function() {
+      $(this).prop("checked", false).trigger("change");
+    })
     // for dates
     if (parentId == 'publication-date-fields') {
       clearDatesFilter();
+      detectDateChange();
     }
   });
 
   // Get :checked on load and create badges
-  $(".sidebar :checked").each(function(){
+  const checked = $(".sidebar :checked");
+  if (checked.length > 0) {
+    if($("#active-filters").is(":hidden")) {
+      $("#active-filters").show();
+    }
```

Comment:
```suggestion
    showActiveFilters();
```

This method was added below, use it here.